common/guestchain/src/manager.rs

```diff
@@ -198,11 +213,19 @@ impl<PK: crate::PubKey> ChainManager<PK> {
             crate::block::Fingerprint::new(&self.genesis, &next_block);
         self.pending_block = Some(PendingBlock {
             fingerprint,
-            next_block,
+            next_block: next_block.clone(),
             signers: Set::new(),
             signing_stake: 0,
         });
         self.candidates.clear_changed_flag();
+        if self.consensus_states.len() == MAX_CONSENSUS_STATES {
+            self.consensus_states.pop_front();
+        }
+        self.consensus_states.push_back(LocalConsensusState {
+            blockhash: next_block.header.calc_hash().to_vec(),
+            height: next_block.block_height,
+            timestamp: next_block.timestamp_ns,
+        });
```

Comment:
If you swap the operations around you don’t need to clone next_block:

```suggestion
        if self.consensus_states.len() == MAX_CONSENSUS_STATES {
            self.consensus_states.pop_front();
        }
        self.consensus_states.push_back(LocalConsensusState {
            blockhash: next_block.header.calc_hash().to_vec(),
            height: next_block.block_height,
            timestamp: next_block.timestamp_ns,
        });
        self.pending_block = Some(PendingBlock {
            fingerprint,
            next_block,
            signers: Set::new(),
            signing_stake: 0,
        });
        self.candidates.clear_changed_flag();
```

Reply:
makes sense.